lib/msal-browser/src/utils/GuidGenerator.ts

```diff
@@ -0,0 +1,78 @@
+     * y could be 1000, 1001, 1010, 1011 since most significant two bits needs to be 10
+     * y values are 8, 9, A, B
+     */
+    static generateGuid(): string {
+        const cryptoObj: Crypto = window.crypto; // for IE 11
+        if (cryptoObj && cryptoObj.getRandomValues) {
+            const buffer: Uint8Array = new Uint8Array(16);
+            cryptoObj.getRandomValues(buffer);
+
+            // buffer[6] and buffer[7] represents the time_hi_and_version field. We will set the four most significant bits (4 through 7) of buffer[6] to represent decimal number 4 (UUID version number).
+            buffer[6] |= 0x40; // buffer[6] | 01000000 will set the 6 bit to 1.
+            buffer[6] &= 0x4f; // buffer[6] & 01001111 will set the 4, 5, and 7 bit to 0 such that bits 4-7 == 0100 = "4".
+
+            // buffer[8] represents the clock_seq_hi_and_reserved field. We will set the two most significant bits (6 and 7) of the clock_seq_hi_and_reserved to zero and one, respectively.
+            buffer[8] |= 0x80; // buffer[8] | 10000000 will set the 7 bit to 1.
+            buffer[8] &= 0xbf; // buffer[8] & 10111111 will set the 6 bit to 0.
+
+            return MathUtils.decimalToHex(buffer[0]) + MathUtils.decimalToHex(buffer[1])
+                + MathUtils.decimalToHex(buffer[2]) + MathUtils.decimalToHex(buffer[3])
+                + "-" + MathUtils.decimalToHex(buffer[4]) + MathUtils.decimalToHex(buffer[5])
+                + "-" + MathUtils.decimalToHex(buffer[6]) + MathUtils.decimalToHex(buffer[7])
+                + "-" + MathUtils.decimalToHex(buffer[8]) + MathUtils.decimalToHex(buffer[9])
+                + "-" + MathUtils.decimalToHex(buffer[10]) + MathUtils.decimalToHex(buffer[11])
+                + MathUtils.decimalToHex(buffer[12]) + MathUtils.decimalToHex(buffer[13])
+                + MathUtils.decimalToHex(buffer[14]) + MathUtils.decimalToHex(buffer[15]);
+        }
+        else {
+            const guidHolder: string = "xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx";
+            const hex: string = "0123456789abcdef";
+            let r: number = 0;
+            let guidResponse: string = "";
+            for (let i: number = 0; i < 36; i++) {
+                if (guidHolder[i] !== "-" && guidHolder[i] !== "4") {
+                    // each x and y needs to be random
+                    r = Math.random() * 16 | 0;
+                }
+                if (guidHolder[i] === "x") {
+                    guidResponse += hex[r];
+                } else if (guidHolder[i] === "y") {
+                    // clock-seq-and-reserved first hex is filtered and remaining hex values are random
+                    r &= 0x3; // bit and with 0011 to set pos 2 to zero ?0??
+                    r |= 0x8; // set pos 3 to 1 as 1???
+                    guidResponse += hex[r];
+                } else {
+                    guidResponse += guidHolder[i];
+                }
+            }
+            return guidResponse;
+        }
+    }
+}
```

Comment:
would be great to be able to use a lib instead of copy these but I know we have had issues there

Reply:
We have had problems in the past trying to make this work with an external library, but if anyone has any suggestions I am open to them.